ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveCalciteUtil.java

```diff
@@ -1257,15 +1249,26 @@ public Void visitCall(RexCall call) {
     }
 
     public boolean hasDisjunction(RexNode node) {
-      // clear the state
-      inNegation = false;
-      hasDisjunction = false;
-
       node.accept(this);
       return hasDisjunction;
     }
   }
 
+  /**
+   * Find disjunction (OR) in an expression (at any level of nesting).
```

Comment:
`Find` implies that we are going to return a disjunctive expression. Suggestion:
`Returns whether the expression has disjunctions (OR) at any level of nesting.`